args: Namespace(seed=42, text_max_len=512, num_train_epochs=5, model_class='CMCModel', dataset_name='ubuntu', label_num=0, composition='pooler', context_num=1, pretrained_bert_path='bert-base-uncased', model_save_prefix='', top_layer_num=3, first_seq_max_len=256, train_candidate_num=-1, used_layers=None, one_stage=False, evaluate_epoch=1, no_aggregator=False, no_enricher=False, no_apex=False, no_train=False, do_test=False, do_real_test=False, query_block_size=200, do_val=False, use_cpu=False, in_batch=False, nvidia_number='1', restore=False, val_batch_size=2, train_batch_size=8, gradient_accumulation_steps=1, step_max_query_num=8, clean_graph=False, no_initial_test=False, only_final=False, load_model=False, load_model_path=None, save_model_dict='./model/', last_model_dict='./last_model/', load_middle=False, distill=False, teacher_path='./model/teacher', mlm=False, local_rank=0, data_parallel=False, data_distribute=False, dataset_split_num=20, first_stage_lr=0.3)
local rank: 0
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ begin one stage train ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
---------------------- create model ----------------------
Traceback (most recent call last):
  File "/home/cme/MixEncoder/nlp_main.py", line 131, in <module>
    my_train_model.train(train_two_stage_flag=my_train_two_stage_flag,
  File "/home/cme/MixEncoder/nlp_trainer.py", line 136, in train
    self.model = self.__create_model()
                 ^^^^^^^^^^^^^^^^^^^^^
  File "/home/cme/MixEncoder/nlp_trainer.py", line 2040, in __create_model
    model = CMCModel(config=self.config)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/cme/MixEncoder/nlp_model.py", line 355, in __init__
    self.extend_multi_transformerencoderlayer = IdentityInitializedTransformerEncoderLayer(self.sentence_embedding_len, self.num_heads, args = args).to(self.device) #TODO: AttributeError!
                                                                                                                                               ^^^^
NameError: name 'args' is not defined
